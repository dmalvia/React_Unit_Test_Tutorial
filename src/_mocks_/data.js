export const usersData = [
  {
    id: "1",
    first_name: "Dipesh",
    email: "dipesh@gmail.com",
    avatar: "dipesh.jpg",
  },
  {
    id: "2",
    first_name: "Nikesh",
    email: "nikesh@gmail.com",
    avatar: "nikesh.jpg",
  },
];

export const badResponse = {
  type: "error",
  title: "Something went wrong",
};
