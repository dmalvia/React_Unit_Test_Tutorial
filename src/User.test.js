import { render, screen } from "@testing-library/react";
import { usersData } from "./_mocks_/data";
import User from "./User";

const renderWithProps = (searchTerm) => {
  return render(<User users={usersData} searchTerm={searchTerm} />);
};

describe("Test User Component", () => {
  it("render users in user component with search term", async () => {
    const { container } = renderWithProps("dipesh");
    const email = await screen.findAllByTestId("email");
    expect(container).toMatchSnapshot();
    expect(email[0].textContent).toBe("dipesh@gmail.com");
    expect(email.length).toEqual(1);
  });
  it("render users in user component with empty search term", async () => {
    const { container } = renderWithProps("");
    const email = await screen.findAllByTestId("email");
    expect(container).toMatchSnapshot();
    expect(email.length).toEqual(usersData.length);
  });
});
