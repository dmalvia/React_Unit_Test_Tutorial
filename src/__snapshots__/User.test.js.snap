// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test User Component render users in user component with empty search term 1`] = `
<div>
  <div
    class="flex"
  >
    <div
      data-testid="users-list"
    >
      <p>
        <strong>
          Dipesh
        </strong>
      </p>
      <p
        data-testid="email"
      >
        dipesh@gmail.com
      </p>
      <img
        alt="1"
        src="dipesh.jpg"
      />
    </div>
    <div
      data-testid="users-list"
    >
      <p>
        <strong>
          Nikesh
        </strong>
      </p>
      <p
        data-testid="email"
      >
        nikesh@gmail.com
      </p>
      <img
        alt="2"
        src="nikesh.jpg"
      />
    </div>
  </div>
</div>
`;

exports[`Test User Component render users in user component with search term 1`] = `
<div>
  <div
    class="flex"
  >
    <div
      data-testid="users-list"
    >
      <p>
        <strong>
          Dipesh
        </strong>
      </p>
      <p
        data-testid="email"
      >
        dipesh@gmail.com
      </p>
      <img
        alt="1"
        src="dipesh.jpg"
      />
    </div>
  </div>
</div>
`;
